web/src/settings_streams.js

```diff
@@ -29,7 +29,8 @@ function get_chosen_default_streams() {
     // Return the set of stream id's of streams chosen in the default stream modal.
     return new Set(
         $("#default-stream-choices .choice-row .dropdown_widget_value")
-            .map((_i, elem) => Number($(elem).attr("data-stream-id")).toString())
+            .not($(".dropdown_widget_value").last())
```

Comment:
Would be good to add a small comment here explaining why we exclude the last element.

Reply:
Sure! Added a note explaining the reason.

Reply:
The selector string inside `.not` is incomplete. This works coincidentally as the last `.dropdown_widget_value` element on the entire page is the one we want here. Going to fix it. Thanks for catching this.

Reply:
Why not just use `.closest/.find` find the element you want?

Reply:
Some alternatives to think about:

- `$("#default-stream-choices .choice-row .dropdown_widget_value").slice(0, -1)`
- `$("#default-stream-choices .choice-row:not(:last-child) .dropdown_widget_value")`
- `$("#default-stream-choices .choice-row .dropdown_widget_value[data-stream-id]")`

Also the comment should explain specifically that we’re excluding the “Select channel” item.

Reply:
`$("#default-stream-choices .choice-row .dropdown_widget_value[data-stream-id]")` seems a lot better to me than skipping the last child, since it's more robust to reordering options. @LaPulgaaa can you use that answer instead and then mark for "integration review"?

Reply:
Yeah I agree.